src/github.js

```diff
@@ -96,20 +96,128 @@ async function fetch_changed_files() {
   return changed_files;
 }
 
+async function is_collaborator(person) {
+  const context = get_context();
+  const octokit = get_octokit();
+
+  return await octokit.repos.checkCollaborator({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    username: person,
+  }).then(
+    (response) => {
+      if (response.status === 204) {
+        return true;
+      }
+      return false;
+    },
+    (error) => {
+      if (error.status === 404) {
+        return false;
+      }
+      throw error;
+    }
+  );
```

Comment:
⚠️ I haven't tested the suggestion below

In JavaScript, when you return a promise, you don't need to add `await`. Also, because we use `async` and `await`, except for some rare cases, you can simply use `async` and `await`:

```suggestion
  try {
    const response = await octokit.repos.checkCollaborator({
      owner: context.repo.owner,
      repo: context.repo.repo,
      username: person,
    });
    
    if (response.status === 204) {
      return true;
    }
    return false;
  } catch (error) {
    if (error.status === 404) {
      return false;
    }
    throw error;
  }
```

Reply:
Thanks for your suggestion, I'll test it later.